### PR TITLE
VSTS DotNet code mirror

### PR DESCRIPTION
**Pull request by mmitche**
Mirror code from github to dotnet.visualstudio.com repos.
Conditionalize variable insertion so that no error is thrown if no variables section is present under phase

### CODE REVIEW

File: .vsts-code-mirror.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Disable CI triggers, only called using Maestro
+trigger: none
+
+# Moves code from GitHub into internal repos
+phases:
+- template: /eng/common/templates/phases/base.yml
+  parameters:
+    agentsOS: Windows NT
```
Comment (chcosta) [inline .vsts-code-mirror.yml]: `Windows_NT`

File: .vsts-code-mirror.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    buildReason: IndividualCI
+    phaseName: Mirror_GitHub_to_VSTS
+    dockerImage: ''
```
Comment (chcosta) [inline .vsts-code-mirror.yml]: you can delete the 'dockerImage' line and 'enableTelemetry' line if you don't need them

**Comment by mmitche** [comment]
@chcosta Any other comments on this?

File: .vsts-code-mirror.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            $vstsRepo = $repo.Replace("/", "-");
+            # Check that the parameters look correct
+            if ($vstsRepo -eq "" -or $branch -eq "")
+            {
+              Write-Error "Expected valid branch and github repo in the form of owner/repo"
+            }
+            Write-Host "##vso[task.setvariable variable=VstsRepoName]$vstsRepo"
+            Write-Host "Mirroring branch '$branch' from github repo '$repo' to vsts repo '$vstsRepo'."
+      - script: |
+          git clone https://dotnet-maestro-bot:$(BotAccount-dotnet-maestro-bot-PAT)@github.com/$(GithubRepo) $(WorkingDirectoryName) -b $(BranchToMirror)
```
Comment (natemcmaster) [inline .vsts-code-mirror.yml]: Did you consider using `--bare`? We don't really need to checkout the source code.

File: .vsts-code-mirror.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Disable CI triggers, only called using Maestro
```
Comment (weshaggard) [inline .vsts-code-mirror.yml]: Do these need to be in the root directory?

**Pull request merged**